[Motwani Richa Vijay] iP

richavm14 opened the pull request:
** Neo**
Neo frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/richavm14/ip/releases/tag/A-Jar).
2. download neo.jar
3. clock it to add your tasks
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [x] Reminders (coming soon)

Commit:
Add Gradle support

richavm14 committed:
changed to neo

richavm14 committed:
added list

richavm14 committed:
mark and unmark

richavm14 committed:
inheritance

richavm14 committed:
testing

richavm14 committed:
exception

richavm14 committed:
delete functionality

richavm14 committed:
adds and retrieves file from Neo.txt

richavm14 committed:
change to Neo.txt

richavm14 committed:
parse date

richavm14 committed:
Merge branch 'branch-Level-7'
add feature store in local file

richavm14 committed:
merge Level-8 resolve conflict

richavm14 committed:
make code follow oop standards

richavm14 committed:
organize into package

richavm14 committed:
Add Update function for storage

richavm14 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

richavm14 committed:
Automate project builds using gradle

richavm14 committed:
add unit tests

L1uY1jun commented on src/main/java/neo/MarkCommand.java:
```diff
@@ -21,5 +21,9 @@ void complete(String tempi) throws NeoException, IOException {
         System.out.println("Nice! I've marked this task as done");
         System.out.println(arrayLL.getTask(tempii-1).toString());
+        stor.Writedata(arrayLL.getTask(0));
+        for(int i =1; i<arrayLL.arrayL.size(); i++) {
```
Convention to have a space around operators based on Java coding standard, this line of code should be written as:
`for (int i = 1; i < arrayLL.arrayL.size(); i++) {`

L1uY1jun commented on src/main/java/neo/UnMarkCommand.java:
```diff
@@ -21,6 +21,10 @@ void complete(String tempi) throws NeoException, IOException {
         System.out.println("ohk I've marked this task as not done");
         System.out.println(arrayLL.getTask(tempii-1).toString());
+        stor.Writedata(arrayLL.getTask(0));
+        for(int i =1; i<arrayLL.arrayL.size(); i++) {
```
Similar issue for this line of code, should be written as: `for (int i = 1; i < arrayLL.arrayL.size(); i++) {` based on java coding standard

L1uY1jun commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,71 @@
+            } catch (NeoException | IOException e) {
+                System.out.println(e.getMessage());
+            }
+        }
```
This section of text could be simplified much more with a 'switch' statement instead of 'if-else' statement
For example:
```java
if (arr.length < 0) {
    throw new NeoException("Sorry I don't know what that means");
} else {
    String command = arr[0];
    switch (command) {
    case "mark": 
        this.mark = new MarkCommand(ui, stor, arrayLL);
        mark.complete(arr[1]);
        break;
    case "unmark":
        this.unmark = new UnMarkCommand(ui, stor, arrayLL);
        unmark.complete(arr[1]);
        break;
    case "delete":
        this.delete = new DeleteCommand(ui, stor, arrayLL);
        delete.complete(arr[1]);
        break;
    case "deadline": 
        //add the other cases and code in
        //below until the default case
    default:
        //throw NeoException if the command string does not match any of the cases
        throw new NeoException("Sorry I don't know what that means");
}
```
read up on the switch statement documentation for more clarifications.

richavm14 replied:
Understood, Will modify it 
Thank you

L1uY1jun commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,9 @@
+        System.out.println("Hello! I'm Neo");
+        System.out.println("What can I do for you?");
+    }
+}
```
The Ui (User interface) class is supposed to handle all the interactions with the user, suggestion for you to add all the `System.out.println(...)` code found in all the *Commands.java classes into this class as methods for Ui to handle everything.

L1uY1jun reviewed (commented):
There was a lot of instances where operators was found not being surrounded by a space character (Java coding standard), for example:
👎 practice: `a=(b+c)*d;`
👍 practice: `a = (b + c) * d;`
same goes for all other operators like `>, <, ||, &&` etc
Overall good effort put into the commits, keep it up!

richavm14 committed:
create jar file

Aloynz commented on src/main/java/neo/AddCommand.java:
```diff
@@ -0,0 +1,49 @@
+import java.io.IOException;
+public class AddCommand extends Command {
+
+    private TaskList arrayLL;
```
Perhaps a more intuitive variable name here?

Aloynz commented on src/main/java/neo/AddCommand.java:
```diff
@@ -0,0 +1,49 @@
+            String temp3 = arri[1];
+
+            Event e = new Event(temp2, temp3);
+            System.out.println("Added: " + e.toString());
+            arrayLL.addTask(e);
+            stor.Storedata(e);
```
Should these line of code be placed after the if statements since it might look repetitive?

Aloynz commented on src/main/java/neo/Deadline.java:
```diff
@@ -0,0 +1,18 @@
+    @Override
+    public String toString() {
+        LocalDate d1 = LocalDate.parse(by);
+        return "[D]" + super.toString() + " (by: " + d1.format(DateTimeFormatter.ofPattern("MMM dd yyyy")) + ")";
```
I like how the toString() method is neatly implemented.

Aloynz commented on src/main/java/neo/Parser.java:
```diff
@@ -0,0 +1,72 @@
+        }
+        else {
+            try {
+                if (!userText.equals("list") && !userText.equals("List") && !arr[0].equals("unmark") && !arr[0].equals("mark") && !arr[0].equals("delete")) {
+                    if (arr.length > 1 && arr[0].equals("deadline")) {
+                        type = 0;
+                        String tempi = arr[1];
+                        this.add = new AddCommand(ui, stor, arrayLL, type);
+                        add.complete(tempi);
+                    }
+                    if (arr.length > 1 && arr[0].equals("event")) {
+                        type = 1;
+                        String tempi = arr[1];
+                        this.add = new AddCommand(ui, stor, arrayLL, type);
+                        add.complete(tempi);
+
+                    }
+                    if (arr.length > 1 && arr[0].equals("todo")) {
+                        type = 2;
+                        String tempi = arr[1];
+                        this.add = new AddCommand(ui, stor, arrayLL, type);
+                        add.complete(tempi);
+                    }
+                    if (arr.length == 1 && arr[0].equals("todo")) {
+                        throw new NeoException("sorry todo cannot be empty");
+                    }
+                    if (!arr[0].equals("todo") && !arr[0].equals("event") && !arr[0].equals("deadline")) {
+                        throw new NeoException("Sorry I don't know what that means");
+                    }
+                }
```
Should switch statement be used in placed of the multiple if statements?

Aloynz reviewed (commented):
Overall, I found the code easy to read except a few places where too many if-else statements were used.  You may wish to consider the use of switch statements in certain areas of the code.

richavm14 committed:
add javadoc comments to code

richavm14 committed:
minor changes

richavm14 committed:
modify code to comply with java coding standards

richavm14 committed:
add functionality to search by keyword

richavm14 committed:
Merge branch 'branch-A-JavaDoc'
add JavaDoc

richavm14 committed:
Merge branch 'branch-A-CodingStandard'

richavm14 committed:
Merge branch 'branch-Level-9'
Add find functionality to allow user to search by keyword

richavm14 committed:
Add more javadoc

zlimez commented on src/main/java/neo/AddCommand.java:
```diff
@@ -0,0 +1,68 @@
+    @Override
+    void complete(String tempi) throws NeoException, IOException {
+
+        if (type ==0) {
+            String arri[];
+            arri = tempi.split("/by ");
+            String temp2 = arri[0];
+            String temp3 = arri[1];
+
+            Deadline d = new Deadline(temp2, temp3);
+            System.out.println("Added: " + d.toString());
+            arrayLL.addTask(d);
+            stor.storeData(d);
+        }
+        if (type ==1) {
+            String arri[];
+            arri = tempi.split("/at ");
+            String temp2 = arri[0];
+            String temp3 = arri[1];
+
+            Event e = new Event(temp2, temp3);
+            System.out.println("Added: " + e.toString());
+            arrayLL.addTask(e);
+            stor.storeData(e);
+        }
+        if (type ==2) {
+            ToDo td = new ToDo(tempi);
+            System.out.println("Added: " + td.toString());
+            arrayLL.addTask(td);
+            stor.storeData(td);
+        }
```
Hmm, since this is also parsing maybe could move this segment of the code to Parser?

zlimez commented on src/main/java/neo/Neo.java:
```diff
@@ -0,0 +1,53 @@
+        Scanner sc = new Scanner(System.in);
+
+        while (true) {
+            System.out.println("Please enter items you want to add to the list, if you want to quit enter bye");
+            System.out.println("");
+            System.out.print("Enter Your command: ");
+            userText = sc.nextLine();
+            System.out.println("");
```
Another thought, maybe this segment can be moved to the UI to make the responsibility of each class narrower?

zlimez reviewed (commented):
Just a few suggestions, and out of curiosity did you implement your gui using javascript jquery haha?

richavm14 committed:
GUI basics

richavm14 committed:
Implement GUI

richavm14 committed:
Merge branch 'branch-Level-10'
Add GUI

richavm14 committed:
The code had instance of deep nesting in parser class in method checkText. The method was very long.

This is not good practice as it decreases the readability of the code

Let us add switch-case statements instead of nested if else statements to prevent this.

richavm14 committed:
Under parser class, the checkText method uses an array to stor elemets of a string after splitting. An assumption is made that the array has more than one element. Complete method in mark and unmark commands don't check if the command has already been marked.

This might lead to error if assumptions made are not true.

Let us add assert statements to avoid possible bugs that can arise from these assumptions.

richavm14 committed:
Merge pull request #2 from richavm14/branch-A-Assertions

Add assert statements

richavm14 committed:
merge master to branch-A-CodeQuality

richavm14 committed:
Merge pull request #3 from richavm14/branch-A-CodeQuality

Improve code quality

richavm14 committed:
add priority to tasks

richavm14 committed:
Merge branch 'branch-C-Priority'
Add functionality to assign priority to task

richavm14 committed:
Fix bug: delete command and exception

richavm14 committed:
Add help command

richavm14 committed:
Merge branch 'branch-C-Help'
Add help functionality

richavm14 committed:
Follows coding standard

richavm14 committed:
Add final modifications

richavm14 committed:
More modifications

richavm14 committed:
Finalise user guide

richavm14 committed:
fix bugs in storage and user guide

richavm14 committed:
update the user guide

richavm14 committed:
Handle more errors

richavm14 committed:
fix test case

richavm14 committed:
Final release

richavm14 committed:
Update the user guide
